includes/compat/subscriptions/trait-wc-payment-gateway-wcpay-subscriptions.php

```diff
@@ -706,4 +706,34 @@ public function restore_order_pay_var() {
 			$wp->query_vars['order-pay'] = $this->order_pay_var;
 		}
 	}
+
+	/**
+	 * Add card mandate options parameters to the order payment intent if needed.
+	 * Only required for the subscription creation for card issued in India.
+	 * More details https://wp.me/pc4etw-ky
+	 *
+	 * @param WC_Order $order The order.
+	 * @return array Params to be included or empty array.
+	 */
+	public function maybe_add_mandate_to_order_payment( WC_Order $order ):  array {
+		$result = [];
+
+		if ( $this->is_subscriptions_enabled() ) {
+			$subscriptions = wcs_get_subscriptions_for_order( $order->get_id() );
+			$subscription  = reset( $subscriptions );
+			if ( $subscription ) {
+				$result['setup_future_usage']                                = 'off_session';
```

Comment:
Appreciate any input here. I think we should apply it to every order with subscriptions, and probably not here.

Reply:
I agree it makes sense to apply it to every Subscription order. As per the Stripe docs:

> You can still accept off-session payments with a card set up for on-session payments, but the bank is more likely to reject the off-session payment and require authentication from the cardholder.

It seems to me that since subscriptions will always have renewal payments, `off_session` makes the most sense here. Maybe we can attach it in a different place, such as when we set up the payment intent? I'm not really sure where the best place for this is.

Reply:
Thanks! I'll need to think more about where to put it. For now, we only set this in `class-wc-payments-api-client.php` if the payment method will be saved. But it isn't enough for this use case. Cause `setup_future_usage=off_session` is always required with e-mandates, and if we use an already saved card will not work.